panic when starting vault server with -recovery flag

**Describe the bug**
```bash
vault server -config assets/vault-cfg.hcl -recovery
```

panics with:

```
panic: flag redefined: combine-logs

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc0010f0a20, {0x7dab0f8, 0xc0012782d0}, {0x6973c02, 0xc}, {0x0, 0x0})
        /home/runner/actions-runner/_work/_tool/go/1.20.3/x64/src/flag/flag.go:982 +0x2f9
github.com/hashicorp/vault/command.(*FlagSet).Var(0xc0012743f0, {0x7dab0f8, 0xc0012782d0}, {0x6973c02, 0xc}, {0x0, 0x0})
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/base_flags.go:857 +0x45
github.com/hashicorp/vault/command.(*FlagSet).VarFlag(0xc0012743f0, 0xc0010bdb70)
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/base_flags.go:807 +0xe5
github.com/hashicorp/vault/command.(*FlagSet).BoolVar(0x194?, 0xc0010bdc18)
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/base_flags.go:156 +0x2a5
github.com/hashicorp/vault/command.(*FlagSet).addLogFlags(0x4f496b?, 0xc000d443c8)
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/log_flags.go:34 +0x85
github.com/hashicorp/vault/command.(*ServerCommand).Flags(0xc000d443c0)
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/server.go:179 +0x1c5
github.com/hashicorp/vault/command.(*ServerCommand).runRecoveryMode(0xc000d443c0)
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/server.go:449 +0x8e
github.com/hashicorp/vault/command.(*ServerCommand).Run(0xc000d443c0, {0xc00011a020, 0x3, 0x3})
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/server.go:935 +0xb05
github.com/mitchellh/cli.(*CLI).Run(0xc0011fc000)
        /home/runner/go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x5f8
github.com/hashicorp/vault/command.RunCustom({0xc00011a010?, 0x4?, 0x4?}, 0xc0000061a0?)
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/main.go:238 +0xa65
github.com/hashicorp/vault/command.Run(...)
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/main.go:142
main.main()
        /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/main.go:16 +0x50
```

**To Reproduce**
1. start the vault server, specify a config file and add the `-recovery` flag

**Expected behavior**
I expect the server to start in recovery mode.

**Environment:**
* Vault Server Version (retrieve with `vault status`): `v.1.13.2`
* Vault CLI Version (retrieve with `vault version`): `Vault v1.13.2+ent (ca5ea376af76440caa5e82e9b659b9e7c09268b4), built 2023-04-25T19:30:45Z`
* Server Operating System/Architecture: `Linux x86_64`

--- a/changelog/20418.txt
+++ b/changelog/20418.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+command/server: fixes panic in Vault server command when running in recovery mode
+```

--- a/command/server.go
+++ b/command/server.go
@@ -457,7 +457,7 @@ func (c *ServerCommand) runRecoveryMode() int {
 	}
 
 	// Update the 'log' related aspects of shared config based on config/env var/cli
-	c.Flags().applyLogConfigOverrides(config.SharedConfig)
+	c.flags.applyLogConfigOverrides(config.SharedConfig)
 	l, err := c.configureLogging(config)
 	if err != nil {
 		c.UI.Error(err.Error())
@@ -671,6 +671,12 @@ func (c *ServerCommand) runRecoveryMode() int {
 	}
 
 	c.UI.Output("")
+
+	// Tests might not want to start a vault server and just want to verify
+	// the configuration.
+	if c.flagTestVerifyOnly {
+		return 0
+	}
 
 	for _, ln := range lns {
 		handler := vaulthttp.Handler.Handler(&vault.HandlerProperties{

--- a/command/server_test.go
+++ b/command/server_test.go
@@ -286,6 +286,13 @@ func TestServer(t *testing.T) {
 			0,
 			[]string{"-test-verify-only"},
 		},
+		{
+			"recovery_mode",
+			testBaseHCL(t, "") + inmemHCL,
+			"",
+			0,
+			[]string{"-test-verify-only", "-recovery"},
+		},
 	}
 
 	for _, tc := range cases {
@@ -295,26 +302,21 @@ func TestServer(t *testing.T) {
 			t.Parallel()
 
 			ui, cmd := testServerCommand(t)
-			f, err := ioutil.TempFile("", "")
-			if err != nil {
-				t.Fatalf("error creating temp dir: %v", err)
-			}
-			f.WriteString(tc.contents)
-			f.Close()
-			defer os.Remove(f.Name())
+
+			f, err := os.CreateTemp(t.TempDir(), "")
+			require.NoErrorf(t, err, "error creating temp dir: %v", err)
+
+			_, err = f.WriteString(tc.contents)
+			require.NoErrorf(t, err, "cannot write temp file contents")
+
+			err = f.Close()
+			require.NoErrorf(t, err, "unable to close temp file")
 
 			args := append(tc.args, "-config", f.Name())
-
 			code := cmd.Run(args)
 			output := ui.ErrorWriter.String() + ui.OutputWriter.String()
-
-			if code != tc.code {
-				t.Errorf("expected %d to be %d: %s", code, tc.code, output)
-			}
-
-			if !strings.Contains(output, tc.exp) {
-				t.Fatalf("expected %q to contain %q", output, tc.exp)
-			}
+			require.Equal(t, tc.code, code, "expected %d to be %d: %s", code, tc.code, output)
+			require.Contains(t, output, tc.exp, "expected %q to contain %q", output, tc.exp)
 		})
 	}
 }

